File: leetcode/232/232.cpp
class MyQueue {
public:
    /** Initialize your data structure here. */
    MyQueue() {

    }
    
    /** Push element x to the back of queue. */
    void push(int x) {
        input.push_front(x);
    }
    
    /** Removes the element from in front of queue and returns that element. */
    int pop() {
        int ret(0);
        if (!output.empty())
        {
            ret = output.front();
            output.pop_front();
            return ret;
        }
        
        transfer();
        
        ret = output.front();
        output.pop_front();
        return ret;
    }
    
    /** Get the front element. */
    int peek() {
        if (!output.empty())
        {
            return output.front();
        }
        
        transfer();
        
        return output.front();
    }
    
    /** Returns whether the queue is empty. */
    bool empty() {
        if (!output.empty())
        {
            return false;
        }
        
        transfer();
        
        return output.empty();
    }
    
    void transfer()
    {
        while (!input.empty())
        {
            output.push_front(input.front());
            input.pop_front();
        }
    }
private:
    std::deque<int> input;
    std::deque<int> output;
};

/**
 * Your MyQueue object will be instantiated and called as such:
 * MyQueue* obj = new MyQueue();
 * obj->push(x);
 * int param_2 = obj->pop();
 * int param_3 = obj->peek();
 * bool param_4 = obj->empty();
 */

File: leetcode/Python/232.py
#!/usr/bin/python3
#-*- coding:utf-8 -*-

class MyQueue:

    def __init__(self):
        """
        Initialize your data structure here.
        """
        self.input = []
        self.output = []
        

    def push(self, x: int) -> None:
        """
        Push element x to the back of queue.
        """
        self.input.append(x)
        

    def pop(self) -> int:
        """
        Removes the element from in front of queue and returns that element.
        """
        if len(self.output) != 0:
            return self.output.pop()
        
        self.__transfer()
        
        return self.output.pop()
        

    def peek(self) -> int:
        """
        Get the front element.
        """
        if len(self.output) != 0:
            return self.output[-1]
        
        self.__transfer()
        
        return self.output[-1]
        

    def empty(self) -> bool:
        """
        Returns whether the queue is empty.
        """
        if len(self.output) != 0:
            return False
        
        self.__transfer()
        
        return len(self.output) == 0
    
    def __transfer(self) -> None:
        """
        Transfer element from input to output
        """
        while len(self.input) != 0:
            self.output.append(self.input.pop())


# Your MyQueue object will be instantiated and called as such:
obj = MyQueue()
obj.push("hello")
obj.push("world")
param_2 = obj.pop()
print(param_2)
param_3 = obj.peek()
print(param_3)
param_4 = obj.empty()
print(param_4)